src/domainobjects/instrument.py

```diff
@@ -1,61 +1,139 @@
 from domainobjects.generatable import Generatable
 from datetime import datetime
+from utils.cache import Cache
+import random
 
 class Instrument(Generatable):
-    
-    def generate(self, record_count, custom_args):
-        config = self.get_object_config()
-        records_per_file = config['max_objects_per_file']
-        file_num = 1
-
-        database = self.get_database()
-        
+    """ Class to generate instruments. Generate method generates a set amount
+    of positions. Other generation methods included where instruments are the
+    only domain object requiring these.
+    """
+
+    def generate(self, record_count, start_id):
+        """ Generate a set number of instruments
+
+        Parameters
+        ----------
+        record_count : int
+            Number of instruments to generate
+        start_id : int
+            Starting id to generate from
+
+        Returns
+        -------
+        List
+            Containing 'record_count' instruments
+        """
+
+        cache = Cache()
```

Comment:
Can we remove the Cache? Lets make (Exchange a dependant domain object for Instrument [Primary Exchange]). An Exchange should have:

Exchange Code
Country of Issue
Currency

and they should all be linked in the databases such that if an instrument is linked to the London Stock Exchange, the curreny is GBP and country of Issue is London etc.